### ISSUE TITLE
[4.x] Intermittent HelloWorldAsyncResponseWithRestRequestTest.checkForAsyncMethodRESTRequestMetric

### ISSUE THREAD

**Issue by romain-grecourt** [helidon-io/helidon#4976]
```
java.lang.AssertionError: 
getAsync count value after invocation
Expected: is <1L>
     but: was <0L>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest.checkForAsyncMethodRESTRequestMetric(HelloWorldAsyncResponseWithRestRequestTest.java:94)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:106)
	at io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest$Proxy$_$$_WeldClientProxy.checkForAsyncMethodRESTRequestMetric(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
```

**Comment by tjquinno** [comment]
Also occurs (at least sometimes) in the new GitHub actions.

**Comment by danielkec** [comment]
Just failed in 4x on top of the fix commit ce733cba0a5196fd855f86ab01a1203dae7c30fa

```
[INFO] Running io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.482 s <<< FAILURE! - in io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest
Error:  io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest.checkForAsyncMethodRESTRequestMetric  Time elapsed: 10.149 s  <<< FAILURE!
java.lang.AssertionError: 
getAsync count value after invocation
Expected: is <1L>
     but: was <0L>
	at io.helidon.common.testing.junit5.MatcherWithRetry.assertThatWithRetry(MatcherWithRetry.java:65)
	at io.helidon.microprofile.metrics.HelloWorldAsyncResponseWithRestRequestTest.checkForAsyncMethodRESTRequestMetric(HelloWorldAsyncResponseWithRestRequestTest.java:95)
```
